include/roboteam_ai/world/Field.h

```diff
@@ -107,18 +121,32 @@ class Field {
     // The center y-coordinate of the field (the y-coordinate that corresponds with the center of the field)
     std::optional<double> centerY;
 
-    // The leftmost x-coordinate of the field (the x-coordinate closest to our goal)
+    // The leftmost x-coordinate of the field which is the lowest x-coordinate value (is a negative value) and is the x-coordinate closest to our goal.
     std::optional<double> leftmostX;
 
-    // The rightmost x-coordinate of the field (the x-coordinate closest to the opponents goal)
+    // The rightmost x-coordinate of the field which is the highest x-coordinate value (is a positive value) and is the x-coordinate closest to the opponents goal.
     std::optional<double> rightmostX;
 
-    // The bottommost y-coordinate of the field (the y-coordinate corresponding to the bottom side of the field)
+    // The bottommost y-coordinate of the field which is the lowest y-coordinate value (is a negative value) and is the y-coordinate corresponding to the bottom side of the field.
     std::optional<double> bottommostY;
 
-    // The uppermost y-coordinate of the field (the y-coordinate corresponding to the upper side of the field)
+    // The uppermost y-coordinate of the field which is the highest y-coordinate value (is a positive value) and is the y-coordinate corresponding to the upper side of the field.
     std::optional<double> topmostY;
 
+    // The x-coordinate of the left penalty line (the penalty line closest to our goal).
+    std::optional<double> leftPenaltyX;
+
+    // The x-coordinate of the right penalty line (the penalty line closest to the opponents goal).
+    std::optional<double> rightPenaltyX;
+
+    /* The highest y-coordinate of the penalty lines (since both penalty lines have the same y-coordinates
+     * at top and bottom, we do not need a seperate variable for the left and right penalty line) */
+    std::optional<double> penaltyTopY;
```

Comment:
This variable is usually unreliable, because in ssl_vision it is often not set in a nice way. What is this different with the upperFieldLine.y anyway?

Reply:
The top of the penalty line is different from the top of the upper field line.

Reply:
It is, but it isn't. as i said this variable is unreliable: it is often not set properly in sslvision (don't be surprised if this value represents 999 meters).

Reply:
Well StopFormation::getStopPositions() is the only function that uses it. First it did some complicated stuff which always returned the top y coordinate of the defence area, but that behaviour should be left unchanged. To cover this problem I can add an additional commentswhich warns future software developers about the dangers of using this value.

Reply:
The penalty line is totally different from the defense area!

Reply:
Okay that means that I fixed all your things, so could you approve it?